python/modules/IcePy/Communicator.cpp

```diff
@@ -328,17 +320,39 @@ communicatorDestroyAsync(CommunicatorObject* self, PyObject* /*args*/)
                 Py_XDECREF(self->wrapper);
                 self->wrapper = nullptr;
 
-                PyObject* err = PyErr_Occurred();
-                if (err)
+                PyObject* setResultArgs = PyTuple_Pack(1, Py_None);
```

Comment:
I would call it `emptyTuple`. You're sure that's the simplest way to create an empty tuple?

Reply:
It is not empty it has 1 argument None.

Reply:
We should remove the arg to the callback. Then we can use an empty tuple - an immortal one with 3.13:
https://docs.python.org/3.13/c-api/object.html#c.Py_CONSTANT_EMPTY_TUPLE

Reply:
I would rather keep the arg, this is the same API as asyncio.Future

https://docs.python.org/3/library/asyncio-future.html#asyncio.Future.set_result

We can pass a lambda callback as @externl suggest